unittests/test_json_serialization.py

```diff
@@ -23,15 +24,90 @@ class TestJsonSerialization:
                         ),
                     ],
-                )
+                ),
+                {
+                    "children": [
+                        {
+                            "token": None,
+                            "tree": {
+                                "children": [{"token": {"type": "INT", "value": "53"}, "tree": None}],
+                                "type": "condition_key",
+                            },
+                        },
+                        {
+                            "token": None,
+                            "tree": {
+                                "children": [
+                                    {
+                                        "token": None,
+                                        "tree": {
+                                            "children": [{"token": {"type": "INT", "value": "1"}, "tree": None}],
+                                            "type": "condition_key",
+                                        },
+                                    },
+                                    {
+                                        "token": None,
+                                        "tree": {
+                                            "children": [{"token": {"type": "INT", "value": "2"}, "tree": None}],
+                                            "type": "condition_key",
+                                        },
+                                    },
+                                ],
+                                "type": "and_composition",
+                            },
+                        },
+                    ],
+                    "type": "or_composition",
+                },
+            )
+        ],
+    )
+    def test_tree_serialization(self, tree: Tree, expected_json_dict: dict):
+        json_string = TreeSchema().dumps(tree)
+        assert json_string is not None
+        actual_json_dict = json.loads(json_string)
+        assert actual_json_dict == expected_json_dict
+        deserialized_tree = TreeSchema().loads(json_data=json_string)
+        assert isinstance(deserialized_tree, Tree)
+        assert deserialized_tree == tree
+
+    @pytest.mark.parametrize(
+        "condition_string, expected_json_dict",
+        [
+            pytest.param(
+                "Muss [2] U ([3] O [4])[901] U [555]",
+                {
+                    "children": [
+                        {
+                            "token": None,
+                            "tree": {
+                                "children": [
+                                    {"token": {"type": "MODAL_MARK", "value": "Muss"}, "tree": None},
+                                    {
+                                        "token": {
+                                            "type": "CONDITION_EXPRESSION",
+                                            "value": " [2] U ([3] O [4])[901] U [555]",
```

Comment:
link

Reply:
What to you mean?

Reply:
it would be nice if you could just use one parse-command and have the condition_expression parsed and dumpable as if it was only a condition_expression.

Reply:
And this is at the moment not possible?
I thought there is already a condition parser: https://github.com/Hochfrequenz/ahbicht/blob/main/src/ahbicht/expressions/condition_expression_parser.py

Reply:
but since it's two different grammars the condition expression is not
parsed by default.

parsing/converting it to a tree on serialization seems wrong to me. I
tried in the post_dump method and then came to the conclusion that
combining the grammars might be the better option. when dumping a tree
we usually don't expect the parsing to take place but just serialize
what has already been parsed

Reply:
ah slowly I get your idea.
So there should be just one grammar which first builds a tree from an ahb expression and evaluates it afterwards in one step?
And the result of this parsing+evaluation should then serialized into json?

Reply:
>  and evaluates it afterwards

**expression evaluates it** / parses it; to not confuse it with the content evaluation

The result should be something which we can e.g. provide to a frontend to make a nice visualization from it

Reply:
But this PR can live without it. It's an improvement anyways imo because we get rid of the "string" keys and name stuff as what it is: a token.